### ISSUE TITLE
Unsupported method: personal_sign[Replace me with meaningful title]

### ISSUE THREAD

**Issue by emmajane1313** [ceramicnetwork/js-did#136]
I'm building a nextjs app that uses wagmi hooks for most of the web3 front end calls. When setting up the ceramic account auth and trying to call

await DIDSession.authorize(authProv, {
        resources: [`ceramic://*`],
        
      })
I keep getting the following error from alchemy upon request:

bad response (status=400, headers={"content-length":"173","content-type":"application/json"}, body="{\"jsonrpc\":\"2.0\",\"id\":42,\"error\":{\"code\":-32600,\"message\":\"Unsupported method: personal_sign. See available methods at https://docs.alchemy.com/alchemy/documentation/apis\"}}

Is there a way to switch from personal_sign to ethers_sign ?

I am getting my signer and provider from wagmi and was previously receiving another bad request error when calling await threeIDConnect.connect(auth) but was able to solve this by using Eip1193Bridge on the wagmi provider response so it could be correctly read by threeId

```
const ethersSigner = useSigner().data as Signer;
  const wagmiProvider = useProvider({
    chainId: 8001,
  });
  const provider = new Eip1193Bridge(ethersSigner, wagmiProvider);
 const auth = new EthereumAuthProvider(provider, address as string);
 const threeIDConnect = new ThreeIdConnect();
 await threeIDConnect.connect(auth)

```
I'm not sure if the above is related to the error I'm currently experiencing but any help on how I can get the personal_signer or circumnavigated is much appreciated :)

**Comment by zachferland** [comment]
Im assuming alchemy doesnt sign transactions/messages? just offers a read api? 

you will need some wallet for users in your app or your server to sign, any wallet will support the personal_sign method for signing messages, typically an injected provider like metamask is used, and it looks like wagmi supports connecting to any of those 

If you want to just sign/create data on your server, you would use did:key without blockchain accounts

**Comment by emmajane1313** [comment]
hey, i figured it out, it was an issue with wagmi provider compatibility,
closing this now :)

**Comment by atulpatare** [comment]
@emmajane1313 How did you solve this? I also not able to get the sign message in wallet to pop up.

My Code
```ts
const {address, isConnected} = useAccount();
const provider = useProvider({
  chainId: mainnet.id,
});

const handleDidSession = () => {
  const accountId = await getAccountId(provider, address);
  const authMethod = await EthereumWebAuth.getAuthMethod(
    provider,
    accountId
  );

  const oneWeek = 60 * 60 * 24 * 7 * 1;
  const session = await DIDSession.authorize(authMethod, {
    resources: [`ceramic://*`],
    expiresInSecs: oneHundredWeeks,
  });
}
```

**Comment by emmajane1313** [comment]
are you using wagmi / rainbowkit? or how are you connecting wallets?

**Comment by zachferland** [comment]
@atulpatare are you getting any errors or logs?

**Comment by AP-Atul** [comment]
@emmajane1313 @zachferland I got this sorted out just now. The `provider` needs to come from `connector` from wagmi's `useAccount` hook.
I have posted the solution here #149

**Comment by jblewnormal** [comment]
Has anyone found success resolving this in a Node server environment (i.e. Lambda)? We've been using a Web3.js implementation on our server side for interacting with some of our smart contracts successfully, but when we pass the Web3 provider into Compose, it continues to throw the `personal_sign` error. Curious if anyone has flushed this out?

**Comment by jblewnormal** [comment]
Hey @zachferland , thanks for the help on this one. I've been using the following (as well as trying others):
```
const web3 =
  new Web3.providers.HttpProvider(
    'https://mainnet.infura.io/v3/<apiKey>'
  );
```
But continually run into the same error. Are there any issues with this provider that you can see top level? Would something like a pure Infura or Alchemy provider work better maybe?

**Comment by zachferland** [comment]
@jblewnormal yeah that would be a read only provider to make chain queries, you would need a wallet provider/signer to be able to sign requests for ceramic

**Comment by jblewnormal** [comment]
> you would need a wallet provider/signer to be able to sign requests for ceramic
@zachferland What's the best one to use in your opinion, because I've tested with both Wagmi and Web3.js and can't seem to get the right setup.
